src/Test/Functional/Acceptance/SplitDbCest.php

```diff
@@ -0,0 +1,279 @@
+     * @throws TaskException
+     */
+    private function partWithoutSplitDbArch(CliTester $I, array $envMagento)
+    {
+        $I->runEceDockerCommand('build:compose --mode=production');
+
+        // Deploy 'Split Db' with the wrong Split Db type
+        $envMagento['stage']['deploy']['SPLIT_DB'] = 'quote';
+        $I->writeEnvMagentoYaml($envMagento);
+
+        $I->runDockerComposeCommand('run build cloud-build');
+        $I->seeInOutput([
+            'ERROR: Fix configuration with given suggestions:',
+            '- Environment configuration is not valid.',
+            'Correct the following items in your .magento.env.yaml file:',
+            'The SPLIT_DB variable contains an invalid value of type string. Use the following type: array.',
+        ]);
+
+        $I->stopEnvironment();
+
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['checkout'];
+        $I->writeEnvMagentoYaml($envMagento);
+        $I->runDockerComposeCommand('run build cloud-build');
+        $I->seeInOutput([
+            'ERROR: Fix configuration with given suggestions:',
+            '- Environment configuration is not valid.',
+            'Correct the following items in your .magento.env.yaml file:',
+            'The SPLIT_DB variable contains the invalid value.',
+            'It should be array with next available values: [quote, sales].'
+        ]);
+
+        $I->stopEnvironment();
+
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['quote', 'checkout'];
+        $I->writeEnvMagentoYaml($envMagento);
+        $I->runDockerComposeCommand('run build cloud-build');
+        $I->seeInOutput([
+            'ERROR: Fix configuration with given suggestions:',
+            '- Environment configuration is not valid.'
+            , 'Correct the following items in your .magento.env.yaml file:',
+            'The SPLIT_DB variable contains the invalid value.'
+            , 'It should be array with next available values: [quote, sales].',
+        ]);
+
+        $I->stopEnvironment();
+
+        // Deploy 'Split Db' with the unavailable Split Db types
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['quote', 'sales'];
+        $I->writeEnvMagentoYaml($envMagento);
+        $this->runDeploy($I);
+        $I->seeInOutput(
+            'ERROR: Enabling a split database will be skipped.'
+            . ' Relationship do not have configuration for next types: sales, quote'
+        );
+        $this->checkEnvPhpConfig($I, [], ['checkout', 'sales']);
+        $this->checkMagentoFront($I);
+    }
+
+    /**
+     * @param CliTester $I
+     * @param array $envMagento
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     * @throws TaskException
+     */
+    private function partWithSplitDbArch(CliTester $I, array $envMagento)
+    {
+        $I->runEceDockerCommand('build:compose --mode=production');
+
+        // Run splitting database for 'quote' tables
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['quote'];
```

Comment:
Looks like these blocks could be split into a helper function

Reply:
Done